### PR TITLE
Fix naive bayes perf

### PR DESCRIPTION
**Pull request by ShvetsKS**

### CODE REVIEW

**Comment by ShvetsKS** [comment]
make test

**Comment by ShvetsKS** [comment]
https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/e9f0d3b9-6350-f1f0-9e5d-8cdcd4b723a1?

File: algorithms/kernel/naivebayes/naivebayes_train_distributed_input.cpp [hunk 1]
```diff
@@ -138,6 +138,49 @@ Status DistributedInput::check(const daal::algorithms::Parameter *parameter, int
...
+    data_management::NumericTablePtr labelsTable = get(classifier::training::labels);
+    DAAL_CHECK_STATUS(s, data_management::checkNumericTable(labelsTable.get(), labelsStr(), 0, 0, 1, nRows));
+
+    data_management::NumericTablePtr weightsTable = get(classifier::training::weights);
+    if (weightsTable)
+    {
+        DAAL_CHECK_STATUS(s, data_management::checkNumericTable(weightsTable.get(), weightsStr(), 0, 0, 1, nRows));
+    }
+
+    if (parameter != NULL)
```
Comment (ntfshard) [inline algorithms/kernel/naivebayes/naivebayes_train_distributed_input.cpp]: nullptr?

File: algorithms/kernel/naivebayes/naivebayes_train_impl.i [hunk 2]
```diff
@@ -107,7 +107,7 @@ struct localDataCollector<algorithmFPType, defaultDense, cpu>
         for( size_t j=0; j<blockSize; j++ )
         {
             int  cl   = predefClass[j];
-
+            DAAL_ASSERT(cl < _c);
           PRAGMA_VECTOR_ALWAYS
```
Comment (ntfshard) [inline algorithms/kernel/naivebayes/naivebayes_train_impl.i]: Could you please align it either

**Pull request merged**